qlib/data/cache.py

```diff
@@ -39,33 +40,123 @@ class QlibCacheException(RuntimeError):
     pass
 
 
-class MemCacheUnit(OrderedDict):
+class MemCacheUnit(abc.ABC):
     """Memory Cache Unit."""
 
-    # TODO: use min_heap to replace ordereddict for better performance
-
     def __init__(self, *args, **kwargs):
-        self.size_limit = kwargs.pop("size_limit", None)
-        # limit_type: check size_limit type, length(call fun: len) or size(call fun: sys.getsizeof)
+        self.size_limit = kwargs.pop("size_limit", 0)
         self.limit_type = kwargs.pop("limit_type", "length")
-        super(MemCacheUnit, self).__init__(*args, **kwargs)
-        self._check_size_limit()
+
+        assert self.limit_type in ["length", "sizeof"], ValueError("limit_type shoule be one of ['length', 'sizeof']")
+        assert self.size_limit >= 0, ValueError("size_limit shoule not be negative.The default 0 means unlimited!")
+
+        # limit_flag: whether to popitem or not
+        self._limit_flag = 1 if self.size_limit > 0 else 0
```

Comment:
I think merge `_limit_flag` into variable `size_limit` will make it simpler
For exmaple,   `size_limit == -1`   indicates no limitation

Reply:
- I think `size_limit == 0` is better because memcache does not make sense when `size_limit <= 0`.